src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/HttpClient/SendAsync.cs

```diff
@@ -97,13 +97,10 @@ public AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall ins
             }
             else
             {
-                // 1.  Since this finishes on a background thread, it is possible it will race the end of
-                //     the transaction. Using holdTransactionOpen = true to prevent the transaction from ending early.
-                // 2.  Do not want to post to the sync context as this library is commonly used with the
-                //     blocking TPL pattern of .Result or .Wait(). Posting to the sync context will result
-                //     in recording time waiting for the current unit of work on the sync context to finish.
-                //     This overload GetAsyncDelegateFor does not use the synchronization context's task scheduler.
-                return Delegates.GetAsyncDelegateFor<Task<HttpResponseMessage>>(agent, segment, true, InvokeTryProcessResponse);
+                // With .Net 6 the HttpClient headers are not thread safe, so we need our continuation to happen synchronously.
```

Comment:
Note that the headers were never thread safe, the race conditions are just easier to hit on 6.0.
Even on 5.0, you may end up corrupting the headers or throwing during enumeration.

Reply:
Good to know. We couldn't reproduce this race condition on 5.0, but this fix does ensure synchronization across all versions. I've updated the comment with a bit more detail.

Reply:
It would only happen in very specific situations - the destination would have to respond with an invalid header that would then be removed during enumeration. You could hit a race there where one thread was removing a value from a dictionary while another attempted to read from it.